### ISSUE TITLE
New version won't complete the randomization of the ROM using chosen settings

### ISSUE THREAD

**Issue by ScienceTynan** [voliol/universal-pokemon-randomizer#3]
This only effects PokeRandoZX-v4.6.0+V0.11.0 while using Pokemon Soul Silver (I have not tested other games)

If I use older versions, it runs fine. I have tried this on both Mac and Windows (using Parallels Desktop), and the same issue persists for me.

These settings apparently cause it to have an error and not complete.

<img width="771" alt="Screenshot 2024-05-02 at 1 56 43 PM" src="https://github.com/voliol/universal-pokemon-randomizer/assets/152658217/c8c57057-ce49-411c-a2df-68f6ac553b08">

It seems there is a conflict with the settings around giving items to trainers and the type table settings. I did not change the new types section, so I'm confused.

It also says the ROM isn't a clean official version but I have used multiple untouched versions, and it says this for all of them.

If any help can be given, I'd appreciate it.

**Comment by voliol** [comment]
Thanks for the report, I'll take a look at it! 
It reporting ROMs of it being non-clean versions is something that has been around for a while, but I haven't dealt with out of uncertainty whether my own ROMs really _were_ clean. The TypeTable issue however is clearly caused by a recent refactor, somehow. 

(Also, if possible, try copy-pasting or attaching the error log, instead of taking a screen shot. The settings string is hard to parse this way ^^)

**Comment by ScienceTynan** [comment]
> Thanks for the report, I'll take a look at it! It reporting ROMs of it being non-clean versions is something that has been around for a while, but I haven't dealt with out of uncertainty whether my own ROMs really _were_ clean. The TypeTable issue however is clearly caused by a recent refactor, somehow.
> 
> (Also, if possible, try copy-pasting or attaching the error log, instead of taking a screen shot. The settings string is hard to parse this way ^^)

Sorry about that, here's the full text:

--

Randomizer Version: 4.6.0 + V0.11.0
Seed: 168566094503019
Settings String: 330IygEfoCYAJsAngAyAEAeglwIagaoarw9CQABAwAAAAAAqAyIMgQyAQAICTIJCQAyVwUAAUQAAEEAFlBva2Vtb24gU291bFNpbHZlciAoVSnwyiOv48M4ig==
Java Version: 21.0.1, Java HotSpot(TM) 64-Bit Server VM
ROM: Pokemon SoulSilver (U)
Code: IPGE
Reported Support Level: Complete

java.lang.RuntimeException: Expected 0xffffffd1, was 0xffffffe0. Likely TMMovesReusableFunctionOffset is faulty.
	at com.dabomstew.pkrandom.romhandlers.Gen4RomHandler.applyReusableTMsPatch(Gen4RomHandler.java:5269)
	at com.dabomstew.pkrandom.romhandlers.Gen4RomHandler.applyMiscTweak(Gen4RomHandler.java:4993)
	at com.dabomstew.pkrandom.romhandlers.AbstractRomHandler.applyMiscTweaks(AbstractRomHandler.java:7091)
	at com.dabomstew.pkrandom.Randomizer.randomize(Randomizer.java:163)
	at com.dabomstew.pkrandom.newgui.NewRandomizerGUI.lambda$performRandomization$132(NewRandomizerGUI.java:1093)
	at java.base/java.lang.Thread.run(Thread.java:1583)

--ROM Diagnostics--
The ROM you loaded is not a clean, official ROM.
Randomizing ROM hacks or bad ROM dumps is not supported and may cause issues.

File name: Medium Random Soul Silver.nds
arm9: 1AB04912
overlay9_12: 68A5B5D8
overlay9_1: 4EEAF0D4
overlay9_21: AC86914B
overlay9_61: B255CB19
a/0/0/1: 6644F919
a/0/0/2: 8A0640C3
a/0/0/4: 373BBE93
a/0/1/1: 23C8D4D3
a/0/1/2: 1DC64A6E
a/0/1/7: 40823EDE
a/0/2/7: A852F4A9
a/0/3/2: 3AC93851
a/0/3/3: 87DF232A
a/0/3/4: 2C4CCFFB
a/0/5/5: D0F3B8A4
a/0/5/6: 4909BD5F
a/1/1/2: 68615C6A
a/1/1/4: EE205002
a/1/3/3: AC3F5797
a/1/3/6: 9B840913
a/2/2/9: 050ECF49
a/2/5/2: A9E8377F
data/mushi/mushi_encount.bin: B980D422
fielddata/maptable/mapname.bin: 5547F5BE
fielddata/wazaoshie/waza_oshie.bin: 664E0F0C
poketool/personal/pms.narc: C457991B

**Comment by voliol** [comment]
Thanks! I've identified the source of the crash, so it should be fixed in the next version. 
If you want to randomize a ROM before then, you'll have to do without "sensible items" and "better movesets" for trainer Pokémon, since that's where the bug can be encountered.
